src/hardware/pcspeaker.cpp

```diff
@@ -73,8 +103,30 @@ static void AddDelayEntry(float index,float vol) {
 		return;
 	}
 	spkr.entries[spkr.used].index=index;
-	spkr.entries[spkr.used].vol=vol;
+
+	const bool is_square_wave =
+	        ((spkr.pit_mode &&
+	          (spkr.prev_mode == SPKR_PIT_ON || spkr.mode == SPKR_PIT_ON)) ||
+	         (!spkr.prev_pos &&
+	          (static_cast<int>(spkr.prev_mode) + static_cast<int>(spkr.mode) ==
+	           SPKR_PIT_ON)));
```

Comment:
This calculation is so complicated, that I think it should be extracted to a new static function here.

Reply:
Yes, reading it again after a couple weeks away from it told me the same thing!
Spelling it out line-by-line has helped me describe it better too.